File: app/src/main/java/sk/styk/martin/apkanalyzer/business/service/ResourceService.java
package sk.styk.martin.apkanalyzer.business.service;

import android.support.annotation.NonNull;

import java.util.HashSet;
import java.util.Set;

import sk.styk.martin.apkanalyzer.model.detail.FileData;
import sk.styk.martin.apkanalyzer.model.detail.FileEntry;
import sk.styk.martin.apkanalyzer.model.detail.ResourceData;

/**
 * Created by Martin Styk on 22.06.2017.
 */

public class ResourceService {

    public ResourceData get(@NonNull FileData fileData) {
        ResourceData data = new ResourceData();

        int numJpg = 0;
        int numGif = 0;
        int numPng = 0;
        int numXml = 0;
        int numNinePatchPng = 0;

        int ldpi = 0;
        int mdpi = 0;
        int hdpi = 0;
        int xhdpi = 0;
        int xxhdpi = 0;
        int xxxhdpi = 0;
        int nodpi = 0;
        int tvdpi = 0;

        Set<String> drawables = new HashSet<>();
        Set<String> layouts = new HashSet<>();

        for (FileEntry file : fileData.getDrawableHashes()) {
            String name = file.getPath();

            // eliminate duplicities
            drawables.add(file.getFileName());

            if (name.endsWith(".jpg")) numJpg++;
            else if (name.endsWith(".gif")) numGif++;
            else if (name.endsWith(".9.png")) numNinePatchPng++;
            else if (name.endsWith(".png")) numPng++;
            else if (name.endsWith(".xml")) numXml++;

            if (name.contains("ldpi")) ldpi++;
            else if (name.contains("mdpi")) mdpi++;
            else if (name.contains("xxxhdpi")) xxxhdpi++;
            else if (name.contains("xxhdpi")) xxhdpi++;
            else if (name.contains("xhdpi")) xhdpi++;
            else if (name.contains("hdpi")) hdpi++;
            else if (name.contains("nodpi")) nodpi++;
            else if (name.contains("tvdpi")) tvdpi++;
            else mdpi++;
        }

        // just adding it to set to eliminate duplicities
        for (FileEntry file : fileData.getLayoutHashes()) {
            layouts.add(file.getFileName());
        }


        data.setDrawables(fileData.getDrawableHashes().size());
        data.setPngDrawables(numPng);
        data.setNinePatchDrawables(numNinePatchPng);
        data.setJpgDrawables(numJpg);
        data.setGifDrawables(numGif);
        data.setXmlDrawables(numXml);
        data.setLdpiDrawables(ldpi);
        data.setMdpiDrawables(mdpi);
        data.setHdpiDrawables(hdpi);
        data.setXhdpiDrawables(xhdpi);
        data.setXxhdpiDrawables(xxhdpi);
        data.setXxxhdpiDrawables(xxxhdpi);
        data.setNodpiDrawables(nodpi);
        data.setTvdpiDrawables(tvdpi);
        data.setLayouts(fileData.getLayoutHashes().size());
        data.setDifferentDrawables(drawables.size());
        data.setDifferentLayouts(layouts.size());

        return data;
    }
}

File: app/src/main/java/sk/styk/martin/apkanalyzer/business/analysis/logic/ResourceService.kt
package sk.styk.martin.apkanalyzer.business.analysis.logic

import sk.styk.martin.apkanalyzer.model.detail.FileData
import sk.styk.martin.apkanalyzer.model.detail.ResourceData
import java.util.*

/**
 * @author Martin Styk
 * @version 22.06.2017.
 */

class ResourceService {

    fun get(fileData: FileData): ResourceData {

        var numJpg = 0
        var numGif = 0
        var numPng = 0
        var numXml = 0
        var numNinePatchPng = 0

        var unspecifiedDpi = 0
        var ldpi = 0
        var mdpi = 0
        var hdpi = 0
        var xhdpi = 0
        var xxhdpi = 0
        var xxxhdpi = 0
        var nodpi = 0
        var tvdpi = 0

        val differentDrawables = HashSet<String>()
        val differentLayouts = HashSet<String>()

        fileData.drawableHashes.forEach {
            val name = it.path
            // eliminate duplicities
            differentDrawables.add(it.fileName)

            when {
                name.endsWith(".jpg") -> numJpg++
                name.endsWith(".gif") -> numGif++
                name.endsWith(".9.png") -> numNinePatchPng++
                name.endsWith(".png") -> numPng++
                name.endsWith(".xml") -> numXml++
            }

            when {
                name.contains("ldpi") -> ldpi++
                name.contains("mdpi") -> mdpi++
                name.contains("xxxhdpi") -> xxxhdpi++
                name.contains("xxhdpi") -> xxhdpi++
                name.contains("xhdpi") -> xhdpi++
                name.contains("hdpi") -> hdpi++
                name.contains("nodpi") -> nodpi++
                name.contains("tvdpi") -> tvdpi++
                else -> unspecifiedDpi++
            }
        }

        // just adding it to set to eliminate duplicities
        fileData.layoutHashes.forEach { differentLayouts.add(it.fileName) }

        return ResourceData(
                pngDrawables = numPng,
                ninePatchDrawables = numNinePatchPng,
                jpgDrawables = numJpg,
                gifDrawables = numGif,
                xmlDrawables = numXml,
                drawables = fileData.drawableHashes.size,
                differentDrawables = differentDrawables.size,
                ldpiDrawables = ldpi,
                mdpiDrawables = mdpi,
                hdpiDrawables = hdpi,
                xhdpiDrawables = xhdpi,
                xxhdpiDrawables = xxhdpi,
                xxxhdpiDrawables = xxxhdpi,
                nodpiDrawables = nodpi,
                tvdpiDrawables = tvdpi,
                unspecifiedDpiDrawables = unspecifiedDpi,
                layouts = fileData.layoutHashes.size,
                differentLayouts = differentLayouts.size)
    }
}
